src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -244,6 +245,15 @@ MultiTaskTrackerExecute(Job *job)
 																task, taskExecution);
 			Assert(execTaskTracker != NULL);
 
+			/* in case the task has intermediate results */
+			if (CheckIfSizeLimitIsExceeded())
+			{
+				failedTaskId = taskExecution->taskId;
+				taskFailed = true;
+				sizeLimitIsExceeded = true;
+				break;
+			}
```

Comment:
If we no longer show a task ID in the `errmsg`, we could move this code below the loop and just do:

```
if (CheckIfSizeLimitIsExceeded())
{
    sizeLimitIsExceeded = true;
}
```

then the `|| sizeLimitIsExceeded` you added to the while condition will make sure we end the loop.